### PR TITLE
api fixed and methods GET and DELETED added for TemplateCombination

### LINKED ISSUE #332: Make TemplateCombination CRUDable
From the api docs, it seem the way to go is directly info, get and delete using the (check list on this post's end). In this case api/template_combinations/ is the base path instead of current api/v2/config_templates/:config_template_id/template_combinations:

GET /api/config_templates/:config_template_id/template_combinations DEPRECATED 	List template combination

1. GET /api/provisioning_templates/:provisioning_template_id/template_combinations 	List template combination
2. GET /api/hostgroups/:hostgroup_id/template_combinations 	List template combination
3. GET /api/environments/:environment_id/template_combinations 	List template combination
4. POST /api/config_templates/:config_template_id/template_combinations DEPRECATED 	Add a template combination
5. POST /api/provisioning_templates/:provisioning_template_id/template_combinations 	Add a template combination
6. POST /api/hostgroups/:hostgroup_id/template_combinations 	Add a template combination
7. POST /api/environments/:environment_id/template_combinations 	Add a template combination
8. GET /api/template_combinations/:id 	Show template combination
9. GET /api/config_templates/:config_template_id/template_combinations/:id DEPRECATED 	Show template combination
10. GET /api/provisioning_templates/:provisioning_template_id/template_combinations/:id 	Show template combination
11. GET /api/hostgroups/:hostgroup_id/template_combinations/:id 	Show template combination
12. GET /api/environments/:environment_id/template_combinations/:id 	Show template combination
13. PUT /api/provisioning_templates/:provisioning_template_id/template_combinations/:id 	Update template combination
14. PUT /api/config_templates/:config_template_id/template_combinations/:id DEPRECATED 	Update template combination
15. PUT /api/hostgroups/:hostgroup_id/template_combinations/:id 	Update template combination
16. PUT /api/environments/:environment_id/template_combinations/:id 	Update template combination
17. DELETE /api/template_combinations/:id 	Delete a template combination

### PR DESCRIPTION
**Pull request by renzon**
fixes #332 

Results:

```python
>>> from nailgun.config import ServerConfig
>>> from nailgun.entities import HostGroup, Environment, ConfigTemplate, TemplateCombination
>>> cfg = ServerConfig('https://foo.bar', 
...                    auth=('baz', '****'), verify=False)
>>> def creat_ent(cls, **kwargs):
...     return cls(cfg, **kwargs)
>>> hostgroup=creat_ent(HostGroup, name='group').create()
>>> env=creat_ent(Environment, name='env').create()
>>> config_template=creat_ent(
...     ConfigTemplate,snippet=False,name='comb',template='cat', template_kind=8,
...     template_combinations=[{'hostgroup_id': hostgroup.id, 'environment_id': env.id}]
... ).create()
>>> combination_dct=config_template.template_combinations[0]
>>> combination_dct
{u'environment_id': 6, u'provisioning_template_name': u'comb', u'environment_name': u'env', u'hostgroup_id': 5, u'config_template_name': u'comb', u'hostgroup_name': u'group', u'provisioning_template_id': 110, u'id': 5, u'config_template_id': 110}
>>> combination=creat_ent(TemplateCombination, id=combination_dct['id'])
>>> combination=combination.read()  # Get working
>>> combination
nailgun.entities.TemplateCombination(nailgun.config.ServerConfig(url='https://foo.bar', verify=False, auth=('baz', '****')), environment=nailgun.entities.Environment(nailgun.config.ServerConfig(url='https://foo.bar', verify=False, auth=('baz', '****')), id=6), config_template=nailgun.entities.ConfigTemplate(nailgun.config.ServerConfig(url='https://foo.bar', verify=False, auth=('baz', '****')), id=110), id=5, hostgroup=nailgun.entities.HostGroup(nailgun.config.ServerConfig(url='https://foo.bar', verify=False, auth=('baz', '****')), id=5))
>>> combination.delete() # Delete working
{u'environment_id': 6, u'provisioning_template_id': 110, u'created_at': u'2016-12-16T12:21:02.687Z', u'updated_at': u'2016-12-16T12:21:02.687Z', u'hostgroup_id': 5, u'id': 5}
```

### CODE REVIEW

**Comment by renzon** [comment]
OBS: New version of nailgun must be publish after this merge

File: nailgun/entities.py [hunk 1]
```diff
@@ -5298,7 +5300,8 @@ def read(self, entity=None, attrs=None, ignore=None):
         return super(System, self).read(entity, attrs, ignore)
 
 
-class TemplateCombination(Entity):
+class TemplateCombination(Entity, EntityReadMixin, EntityDeleteMixin,
```
Comment (oshtaier) [inline nailgun/entities.py]:
```
class TemplateCombination(
        Entity,
        EntityDeleteMixin 
        EntityReadMixin,
        EntityUpdateMixin):
```

**Comment by oshtaier** [comment]
Yeah, only get and delete methods are present using that base URL

**Comment by oshtaier** [comment]
Please, use direct nailgun calls for testing and providing results. Robotello is redundant layer in that case

**Comment by renzon** [comment]
@oshtaier normally I would use directly nailgun. But [robottelo tes](https://github.com/SatelliteQE/robottelo/pull/4099/files)t was already done. In fact it was the reason I found those bugs on nailgun. So I took the lazy path and just posted the tests using it =P. But if you think directly call is really needed, I can make it ;)

**Comment by renzon** [comment]
I don't know why is travis failing on docs creation only on Python 3.3.

**Comment by svtkachenko** [comment]
@renzon If I remember correctly it's because latest sphinx requires python 3.4+

File: nailgun/entities.py [hunk 2]
```diff
@@ -5298,7 +5300,7 @@ def read(self, entity=None, attrs=None, ignore=None):
         return super(System, self).read(entity, attrs, ignore)
 
 
-class TemplateCombination(Entity):
+class TemplateCombination(Entity, EntityReadMixin, EntityDeleteMixin):
```
Comment (svtkachenko) [inline nailgun/entities.py]: I think usually we put mixins in alphabetical order.

Comment (renzon) [inline nailgun/entities.py]: Fixed ;)

Comment (oshtaier) [inline nailgun/entities.py]: already wrote about that above:)

**Comment by oshtaier** [comment]
Direct call will instantly show whether every mixin works and you really can perform all actions that you like

**Comment by oshtaier** [comment]
About travis - yes, we had discussion with @omaciel and @rochacbruno about that, but we didn't have any final decision about that. We have to whether remove 3.3 verification at all or make sphinx static for version when it worked for 3.3

File: nailgun/entities.py [hunk 3]
```diff
@@ -5298,7 +5300,7 @@ def read(self, entity=None, attrs=None, ignore=None):
         return super(System, self).read(entity, attrs, ignore)
 
 
-class TemplateCombination(Entity):
+class TemplateCombination(EntityDeleteMixin, EntityReadMixin, Entity):
```
Comment (oshtaier) [inline nailgun/entities.py]: `TemplateCombination(Entity, EntityDeleteMixin, EntityReadMixin):`

**Comment by renzon** [comment]
@oshtaier I edited https://github.com/SatelliteQE/nailgun/pull/342#issue-196039063 to have nailgun session

**Comment by renzon** [comment]
@oshtaier @svtkachenko Entities Mixings order fixed

**Comment by renzon** [comment]
@oshtaier please check the nailgun session and changes, merging if you think it is ok. Thanks.

**Pull request merged**